### PR TITLE
use pooled files, enable create, make old code generator work

### PR DESCRIPTION
**Pull request by holiman**
This PR fixes some bugs I ran into trying to get it to work with the legacy code generators, and also has some improvements. 

* The fuzzer now does not create/delete files all the time , but instead overwrites old files using a pooled approach. 
* The fuzzer now makes testcases that are create-transactions. 
* The spurious failures #102 is fixed, we simply ignore it if it happens. 
* Added back the old prestate accounts, they had gone missing if we weren't using the smart analyzer to figure out which accounts were used. 

@tintinweb I probably screwed up with your code, sorry

### CODE REVIEW

**Comment by tintinweb** [comment]
@holiman - yeah, this needs to be merged after #116 :)

File: evmlab/tools/statetests/templates/statetest.py [hunk 1]
```diff
@@ -28,11 +28,10 @@ def __dict__(self):
 
 class StateTestTemplate(object):
 
-    def __init__(self, nonce=None, codegenerators={}, datalength=None, fill_prestate_for_tx_to=True, fill_prestate_for_args=False, _config=None):
+    def __init__(self, nonce=None, codegenerators={}, datalength=100, fill_prestate_for_tx_to=True, fill_prestate_for_args=False, _config=None):
```
Comment (tintinweb) [inline evmlab/tools/statetests/templates/statetest.py]: we should probably keep this at None and change the default (should be configurable in ini)

File: evmlab/tools/statetests/templates/statetest.py [hunk 2]
```diff
@@ -117,6 +122,8 @@ def _autofill_prestate(self, address):
 
     def _autofill_prestates_from_stack_arguments(self):
         # todo: hacky hack
+        if not rndval.RndCodeSmart2 in self._codegenerators.keys():
```
Comment (tintinweb) [inline evmlab/tools/statetests/templates/statetest.py]: better fix in #116

File: utilities/fuzzer.py [hunk 3]
```diff
@@ -15,6 +15,29 @@
...
+            return self.frees.pop()
+        n = "pool_%d" % self.counter
+        self.counter = self.counter+1
+        return n
+
+    def put(self, f):
+        self.frees.append(f)
+
+
+fPool= FilePool()
```
Comment (tintinweb) [inline utilities/fuzzer.py]: avoid globals. lets have this live inside a class. I'll manually merge this after #116

File: utilities/fuzzer.py [hunk 4]
```diff
@@ -488,17 +516,27 @@ def generate_tests(self):
...
+        ##### Statetest Template init
+        from evmlab.tools.statetests.templates import statetest
 
         # We'll offload test generation to another thread
         q = queue.Queue(maxsize = 20)
         def createATest():
             counter = 0
             while True:
-                test_obj = self.statetest_template.fill()
+                statetest_template = statetest.StateTestTemplate(nonce="0x1d",
```
Comment (tintinweb) [inline utilities/fuzzer.py]: better fix in #116 
https://github.com/ethereum/evmlab/pull/116/files#diff-69770162e2ecacf29d6bf49c29b8eef2R518

File: utilities/fuzzer.py (hunk 3 above)
Comment (holiman) [reply utilities/fuzzer.py]: Generally I agree, for pools however, might be ok to use globals

**Comment by holiman** [comment]
Rebased on yours, and fixed some other things. I'll merge this and we can fix it up on head instead.

**Pull request merged**